folium/features.py

```diff
@@ -1212,7 +1213,14 @@ def __init__(self, geo_data, data=None, columns=None, key_on=None,  # noqa
         if color_data is not None and key_on is not None:
             real_values = np.array(list(color_data.values()))
             real_values = real_values[~np.isnan(real_values)]
-            _, bin_edges = np.histogram(real_values, bins=bins)
+            if use_jenks:
+                from jenkspy import jenks_breaks
+
+                if not isinstance(bins, int):
+                    raise ValueError(f'bins value must be an integer. Invalid value "{bins}" received.')
+                bin_edges = np.array(jenks_breaks(real_values, bins))
```

Comment:
I got an error with the color lookup because the last value in the array was not inclusive. Making it an array of floats solved the issue. I was using the `main` version of branca and your example notebook.

```suggestion
                bin_edges = np.array(jenks_breaks(real_values, bins), dtype=float)
```